### ISSUE TITLE
getLocation and onLocationUpdate never return any response

### ISSUE THREAD

**Issue by huzain07** [Lyokone/flutterlocation#404]
**Describe the bug**
Location not update on function onLocationChange and getLocation never return any response

**Expected behavior**
get initial location and get location every location changed.

**Tested on:**
 - Android, API Level 26, simulator

**Other plugins:**
nothing

**Additional logs**
this is my code, 
```
@override
  void initState() {
    super.initState();
    isTapped = false;

    location = new Location();

    location.onLocationChanged.listen((LocationData locationData) {
      debugPrint("Update Data");
    });
    setInitialLocation();
  }
```

```
void setInitialLocation() async {
    // set the initial location by pulling the user's
    // current location from the location's getLocation()
    debugPrint("getting location");
    _serviceEnabled = await location.serviceEnabled();
    debugPrint("getting SERVICE : "+_serviceEnabled.toString());
    if (!_serviceEnabled) {
      _serviceEnabled = await location.requestService();
    }

    _permissionGranted = await location.hasPermission();
    if (_permissionGranted == PermissionStatus.denied) {
      debugPrint("getting permission : denied");
      _permissionGranted = await location.requestPermission();
      if (_permissionGranted != PermissionStatus.granted) {
        //return;
      }
    }else{
      debugPrint("getting permission : granted");
    }

    currentLocation = await location.getLocation();
    debugPrint("current location : "+currentLocation.latitude.toString()+","+currentLocation.longitude.toString());
  }
```

service is enabled, and permission is granted but still nothing return
````
I/flutter ( 5601): getting SERVICE : true
I/flutter ( 5601): getting permission : granted
D/        ( 5601): HostConnection::get() New Host Connection established 0x84da7980, tid 5900
D/EGL_emulation( 5601): eglCreateContext: 0x7df3e840: maj 1 min 0 rcv 1
D/EGL_emulation( 5601): eglMakeCurrent: 0xacb3ee60: ver 3 1 (tinfo 0xb0fecf40)
D/EGL_emulation( 5601): eglMakeCurrent: 0x7df3e840: ver 1 0 (tinfo 0x8c0fc3a0)
I/zygote  ( 5601): NativeAllocBackground concurrent copying GC freed 6087(250KB) AllocSpace objects, 0(0B) LOS objects, 49% free, 3MB/7MB, paused 6.493ms total 37.
````

**Comment by yashpaneliya** [comment]
Same issue, @Lyokone please look into this.

**Comment by vicenterusso** [comment]
same issue

**Comment by huzain07** [comment]
> This issue has been automatically marked as stale because it has not had recent activity. It will be closed if no further activity occurs. Thank you for your contributions.

there is no follow up from the dev.

**Comment by asimolmez** [comment]
same issue

**Comment by andrea689** [comment]
any news?

**Comment by huzain07** [comment]
> Any news?
> 
> In the console I received `W/LocationUpdateReceiver( 9573): LocationEngineResult == null`

nope, it's better to change library.

**Comment by esmaeil-ahmadipour** [comment]
in my case , after turn-off my device and restarting , get answer !

**Comment by rubenvereecken** [comment]
I've had similar issues. We used to have a fully working version, then I changed some things and then this started happening, leading me to figure out more or less what's going on.

First off `location.onLocationChanged` is not just an innocent Stream, according to the docs: 
  > It will try to request permission if not granted yet and will throw a PERMISSION_DENIED error code if permission still not granted.

I request permission first and only then set up the listener because I prefer having control over when/where permissions are requested. 

So I changed the flow to:
1. request permission
2. set up listener
And got this strange bug where the listener never even gets set up because `location.onLocationChanged` hangs itself. This is buggy behavior.

Strangely, I managed to work around it by:
1. requesting permission
2. fetching location once
3. setting up listener

@huzain07 I suggest you re-open this issue. I also suggest that `location.onLocationChanged` doesn't do any magic permission requesting, or whatever else makes it an asynchronous Stream getter that can hang.
